### PR TITLE
add sections on home-on-scratch + /readonly mount point to Hortense documentation

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: source/gent/tier1_hortense.rst [hunk 1]
```diff
@@ -216,28 +221,97 @@ This storage space can be addressed with the environment variable $TMPDIR
...
+
+Accessing data via ``/readonly``
+********************************
+
+Due to the fairly aggressive page cache purging policy that is used by the `Lustre <https://www.lustre.org>`_
+storage software that is used for the Tier-1 Hortense scratch filesystem, you may need to make some changes
+to how you access data in your job scripts to avoid performance problems.
+
+Whether or not this is required depends on the specific characteristics of your workloads:
+the number of files that are read, how large those files are, how the data in these files is accessed
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:283]: it is only relevant for data that is only read more than once. for such data, the perf impact depends on number of files, size, io pattern.

File: source/gent/tier1_hortense.rst [hunk 2]
```diff
@@ -216,28 +221,97 @@ This storage space can be addressed with the environment variable $TMPDIR
...
+(the I/O pattern), etc. Note that this applies to both input data for your workloads, as well as
+any software you have installed on the Tier-1 Hortense scratch filesystem (see also :ref:`hortense_software_readonly`).
+
+To mitigate performance problems, you can access the data in your project scratch directory
+through the ``/readonly`` mount point, rather than accessing it directly.
+**We recommend to access your data via the** ``/readonly`` **mount point whenever possible,
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:289]: i would not write this. as it is not easy to determine this. people should try and see if it matters.
we will most likely also tune lustre further so this even less an issue (the aggressive cache flushing can be mitigated somewhat)

File: source/gent/tier1_hortense.rst [hunk 3]
```diff
@@ -216,28 +221,97 @@ This storage space can be addressed with the environment variable $TMPDIR
...
+since it can result in a signficant performance speedup for your jobs.**
+
+This is done by prefixing the path to files and directories with ``/readonly/`` in your job script:
+rather than accessing your data via ``$VSC_SCRATCH_PROJECTS_BASE/...`` (or ``/dodrio/scratch/...``,
+which you should not used), you just use ``/readonly/$VSC_SCRATCH_PROJECTS_BASE/...`` instead.
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:294]: used -> use

File: source/gent/tier1_hortense.rst [hunk 4]
```diff
@@ -216,28 +221,97 @@ This storage space can be addressed with the environment variable $TMPDIR
...
+As the name suggests, the ``/readonly`` mount point only provides *read-only* access to your data.
+Trying to make any changes to files accessing via ``/readonly`` will result in ``Permission denied`` errors.
+
+.. note::
+
+   On the login nodes, there is a delay of maximum 30 minutes for changes to files (or new/removed
+   files/directories) to be reflected through the ``/readlonly/`` mount point.
+
+   In jobs, any changes you make to files or directories in your project scratch directory should be reflected
+   through the ``/readonly`` mount point immediately, as long as the job started running *after* the
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:312]: the immediately is confusing, remove it.

File: source/gent/tier1_hortense.rst [hunk 5]
```diff
@@ -216,28 +221,97 @@ This storage space can be addressed with the environment variable $TMPDIR
...
+   changes were made.
+
+   In addition, take into account that changes which are made while the job is running will *not* be
+   reflected through the ``/readonly`` mount point (for that job).
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:316]: for that -> during that

File: source/gent/tier1_hortense.rst [hunk 6]
```diff
@@ -216,28 +221,97 @@ This storage space can be addressed with the environment variable $TMPDIR
...
+   If your job script creates new files, updates existing files, etc., those changes will *not* be
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:317]: new files/dirs might show up; but people should not assume so. not sure how to phrase this. i don;t mind making it thi strict, but it's not 100% true

File: source/gent/tier1_hortense.rst [hunk 7]
```diff
@@ -216,28 +221,102 @@ This storage space can be addressed with the environment variable $TMPDIR
...
+or in case of problems with the network connection to the other VSC sites.
+
+Although your VSC home directory is usually accessible via ``$VSC_HOME``,
+we strongly recommend to *not* simply create symbolic links to files like your ``.bashrc`` startup script,
+since that would defeat the purpose of this "home-on-scratch" setup.
+
+This recommendation also applies to ``$VSC_DATA``: you should avoid using it in your job scripts as much as
+possible, and ensure that your workflow only relies on the Hortense scratch filesystem. If you require any
+data as input for your jobs, it should be copied to the Hortense scratch filesystem first (for example
+via :ref:`globus`).
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:243]: i would not mention only globus; it's overkill for small things and eg industry people don't have access to it

File: source/gent/tier1_hortense.rst [hunk 8]
```diff
@@ -216,28 +221,100 @@ This storage space can be addressed with the environment variable $TMPDIR
...
+
+.. _hortense_project_scratch_dirs:
+
+Project scratch directories
+***************************
+
+* Please be aware that storage space on ``$VSC_SCRATCH`` (personal scratch directory) is limited per user to 3 GB.
+* Instead, it is better to use the dedicated scratch storage space which is reserved for your Tier-1 project.
+* The environment variable ``$VSC_SCRATCH_PROJECTS_BASE`` points to the base folder containing all project directories.
+* Project directories are given the same name as your Tier-1 project (so *without* a prefix like ``gpr_compute_``).
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:252]: @epauwels what other prefixes do we have : startin and explor?

File: source/gent/tier1_hortense.rst [hunk 9]
```diff
@@ -383,6 +460,45 @@ and load one or more modules via the ``module load`` command to start using them
...
+for example in scripts or binaries that make part of the installation,
+you should install the software using the ``bwrap`` utility. This allows you to "rename" the path to your
+project scratch directory so it starts with ``/readonly/``, while preserving write access to it.
+
+Assuming that the procedure to install the software is implemented in a script named ``install.sh``,
+you can use ``bwrap`` as follows:
+
+.. code::
+
+   bwrap --bind / / --bind $VSC_SCRATCH_PROJECTS_BASE /readonly/$VSC_SCRATCH_PROJECTS_BASE --dev /dev --bind /dev/log /dev/log ./install.sh
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:486]: yuck. can't we make a wrapper for this?

File: source/gent/tier1_hortense.rst (hunk 8 above)
Comment (boegel) [reply source/gent/tier1_hortense.rst:252]: I'm happy to fix the docs to mention all possible prefixes, but not in this PR please :)

File: source/gent/tier1_hortense.rst [hunk 10]
```diff
@@ -383,6 +460,45 @@ and load one or more modules via the ``module load`` command to start using them
...
+
+The ``install.sh`` script should be implemented such that it installs the software to
```
Comment (stdweird) [inline source/gent/tier1_hortense.rst:488]: you can also suggest to start a shell with bwrap, and then install the software from the shell

File: source/gent/tier1_hortense.rst (hunk 9 above)
Comment (boegel) [reply source/gent/tier1_hortense.rst:486]: Yeah, we should, I agree. A commando like `install-in-readonly` or something?

Comment (boegel) [reply source/gent/tier1_hortense.rst:486]: Or `run-in-readonly`

Comment (stdweird) [reply source/gent/tier1_hortense.rst:486]: in the example, i would also add the home `--bind /readonly/$HOME $HOME`

File: source/gent/tier1_hortense.rst (hunk 10 above)
Comment (boegel) [reply source/gent/tier1_hortense.rst:488]: I'd say that's harder to explain... Also, we don't want to encourage people to install stuff manually.

File: source/gent/tier1_hortense.rst (hunk 9 above)
Comment (boegel) [reply source/gent/tier1_hortense.rst:486]: `dodrio-bind-readonly` is now in place, which binds `/dodrio` to `/readonly/dodrio`, so no need to also bind `$HOME` (which is `/dodrio/scratch/users`)

File: source/gent/tier1_hortense.rst (hunk 10 above)
Comment (boegel) [reply source/gent/tier1_hortense.rst:488]: fixed in 9ceb1e7

File: source/gent/tier1_hortense.rst (hunk 9 above)
Comment (boegel) [reply source/gent/tier1_hortense.rst]: fixed in 9ceb1e7

**Review by stdweird — APPROVED** [review]

**Pull request merged**